opentracing/src/main/java/io/opentracing/Span.java

```diff
@@ -52,6 +57,14 @@
    * Returns null if no entry found, or baggage is not supported in the current implementation.
    */
   String getBaggageItem(String key);
+  
+  /**
+   * Useful for acquiring the set of baggage keys so that clients of the Span have the ability
+   * to access the corresponding baggage value.
+   * 
+   * @return {@link Set} of baggage keys
+   */
+  Set<String> getBaggageKeys();
```

Comment:
@toaler @michaelsembwever 

An *implementation* of `Tracer.inject` will be free to cast the `Span` instance to the actual non-interface type, at which point gaining access to the baggage is trivial. I.e., this shouldn't be needed in the `Span` interface, regardless of #20.